File: cookbook/python/documents/extract_api.py
# ---------------------------------------------
## Quick example: Extract structured data using `client.extractions.create()`.
# ---------------------------------------------

import json
import os

from dotenv import load_dotenv

from retab import Retab

# Load environment variables
load_dotenv()

retab_api_key = os.getenv("RETAB_API_KEY")
assert retab_api_key, "Missing RETAB_API_KEY"

# Retab Setup
client = Retab(api_key=retab_api_key)

# Document Extraction via Retab API.
# create() takes keyword args directly (no ExtractionRequest wrapper); the
# document may be a path/bytes/URL, but json_schema must be a dict.
response = client.extractions.create(
    document="../../../assets/docs/invoice.jpeg",
    json_schema=json.load(open("../../../assets/code/invoice_schema.json")),
    model="retab-small",
    n_consensus=1,
)

# Output
print("\n✅ Extracted Result:")
print(response.output)

File: cookbook/node/documents/extract_api.js
// ---------------------------------------------
// Quick example: Extract structured data using Retab's all-in-one `.parse()` method.
// ---------------------------------------------

import { Retab } from '@retab/node';
import { config } from 'dotenv';

// Load environment variables
config();

const retabApiKey = process.env.RETAB_API_KEY;
if (!retabApiKey) {
  throw new Error('Missing RETAB_API_KEY');
}

// Retab Setup
const reclient = new Retab({ api_key: retabApiKey });

// Document Extraction via Retab API
const response = await reclient.documents.extract({
  document: '../../assets/docs/invoice.jpeg',
  model: 'gpt-4.1',
  json_schema: '../../assets/code/invoice_schema.json',
  modality: 'native',
  image_resolution_dpi: 96,
  browser_canvas: 'A4',
  temperature: 0.0,
  n_consensus: 1,
});

// Output
console.log('\n✅ Extracted Result:');
console.log(response.choices[0].message.content);